src/binary/timestamp.rs

```diff
@@ -0,0 +1,98 @@
+// Copyright Amazon.com, Inc. or its affiliates.
+
+use std::io::{self, Write};
+
+use bytes::BufMut;
+use chrono::{Datelike, Timelike};
+use smallvec::SmallVec;
+
+use crate::{
+    binary::{uint::DecodedUInt, var_int::VarInt, var_uint::VarUInt, writer::MAX_INLINE_LENGTH},
+    result::IonResult,
+    types::timestamp::{Precision, Timestamp},
+};
+
+/// Extends [`Timestamp`] to support [Ion binary].
+///
+/// [Ion binary]: https://amzn.github.io/ion-docs/docs/binary.html#6-timestamp
+impl Timestamp {
+    const BUFFER_SIZE: usize = 16;
+
+    /// NOTE: Currently, this function always encodes with nanosecond precision.
```

Comment:
This comment originated on the `write_datetime` method:

https://github.com/amzn/ion-rust/blob/0e3d46ee0b115c496a5192a0b2bb124c8717a9d7/src/binary/writer.rs#L521-L524

It made a bit more sense there because a `DateTime<FixedOffset>` always has all of the time unit fields up to and including fractional seconds.

However, `Timestamp`s can have a precision of `Year` or `Month`, so I think we need a bit of clarification:

```suggestion
    /// NOTE: If the Timestamp being encoded has fractional seconds, they will be written as nanoseconds.
```